Exiled.Events/Patches/Events/Scp079/ChangingCamera.cs

```diff
@@ -7,67 +7,95 @@
+
     /// <summary>
     /// Patches <see cref="Scp079PlayerScript.CallCmdSwitchCamera(ushort, bool)"/>.
     /// Adds the <see cref="Scp079.ChangingCamera"/> event.
     /// </summary>
     [HarmonyPatch(typeof(Scp079PlayerScript), nameof(Scp079PlayerScript.CallCmdSwitchCamera))]
     internal static class ChangingCamera
     {
-        private static bool Prefix(Scp079PlayerScript __instance, ushort cameraId, bool lookatRotation)
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
         {
-            try
-            {
-                if (!__instance._interactRateLimit.CanExecute(true))
-                {
-                    return false;
-                }
-
-                if (!__instance.iAm079)
-                {
-                    return false;
-                }
-
-                Camera079 camera = API.Features.Map.GetCameraById(cameraId);
-
-                if (camera == null)
-                {
-                    return false;
-                }
-
-                float num = __instance.CalculateCameraSwitchCost(camera.transform.position);
-                bool isAllowed = num <= __instance.curMana;
-                ChangingCameraEventArgs ev = new ChangingCameraEventArgs(API.Features.Player.Get(__instance.gameObject), camera, num, isAllowed);
-                Scp079.OnChangingCamera(ev);
-                if (ev.IsAllowed)
-                {
-                    __instance.RpcSwitchCamera(ev.Camera.cameraId, lookatRotation);
-                    __instance.Mana = Mathf.Clamp(__instance.Mana - ev.AuxiliaryPowerCost, 0, __instance.maxMana);
-                    __instance.currentCamera = ev.Camera;
-                }
-                else if (ev.AuxiliaryPowerCost > __instance.curMana)
-                {
-                    __instance.RpcNotEnoughMana(ev.AuxiliaryPowerCost, __instance.curMana);
-                }
-
-                return false;
-            }
-            catch (Exception e)
+            var newInstructions = ListPool<CodeInstruction>.Shared.Rent(instructions);
+
+            // The index offset.
+            const int offset = 0;
+
+            // Search for the first "ldloc.1".
+            var index = newInstructions.FindIndex(instruction => instruction.opcode == OpCodes.Ldloc_1) + offset;
+
+            // Define the first label of the last "ret" and retrieve it.
+            var returnLabel = newInstructions[newInstructions.Count - 1].WithLabels(generator.DefineLabel()).labels[0];
+
+            // Declare a local variable of the type "ChangingCameraEventArgs";
+            var changingCameraEv = generator.DeclareLocal(typeof(ChangingCameraEventArgs));
+
+            // var ev = new ChangingCameraEventArgs(Player.Get(this.gameObject), camera, num,  num <= this.curMana);
+            //
+            // Handlers.Scp079.OnChangingCamera(ev);
+            //
+            // if (!ev.IsAllowed)
+            //   return;
```

Comment:
I see a lack of `// num = ev.AuxiliaryPowerCost` which was confusing me about the local variable.